### PR TITLE
feat(designer): Validate empty required actions inputs prior saving

### LINKED ISSUE #2677: Workflow save button is active when an action is added and there is no user interaction inside the action.
## What is the issue?
Designer in portal and vscode doesn't disable the save button/allows to save when an action is added to the workflow and there is no user interaction in the action input fields.
Clicking in the inputs, not doing anything and clicking outside triggers the error text in the action and the disable of the save button.
Need to come up with an implementation to solve this.

### Describe the Bug with repro steps
1.Create a Stateful workflow in the VS Code project.
2.Open the Designer and add http trigger.
3.Add Office Send an email action.
4.Remain mandatory (*) fields empty without user interaction and try to save the workflow.

Expected: Workflow should not be saved.

Actual: Able to save the workflow.

### What type of Logic App Is this happening in?

Standard Portal and VSCode

### Screenshots or Videos

image

AB#24191431

### PR DESCRIPTION
**Pull request by ccastrotrejo**
* [X] The commit message follows our guidelines
* [X] Tests for the changes have been added (for bug fixes/features)
* [ ] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixes #2677 

- **What is the current behavior?** (You can also link to an open issue here)
Designer doesn't check for required inputs in actions, therefore user can save action that aren't completed

- **What is the new behavior (if this is a feature change)?**
The intention is to make designer check the required inputs for all the actions. This is now checked when clicking in the saving icon as the user will want to complete the workflow. If there are incomplete actions the workflow will not be saved and the error icons will be shown in the respective actions and inputs.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

- **Please Include Screenshots or Videos of the intended change**:

https://github.com/Azure/LogicAppsUX/assets/102700317/c8e4bc89-98ee-4086-be35-5c13eeaf4fab

AB#24656603

### CODE REVIEW

**Review by rllyy97 — APPROVED** [review]
Looks good, will you be adding this to Portal as well?

File: apps/designer-standalone/src/app/AzureLogicAppsDesigner/DesignerCommandBar.tsx [hunk 1]
```diff
@@ -53,9 +55,26 @@ export const DesignerCommandBar = ({
...
+          if (validationErrors.length > 0) {
+            dispatch(updateParameterValidation({ nodeId: id, groupId: parameterGroup.id, parameterId: parameter.id, validationErrors }));
+          }
+          return validationErrors.length;
+        });
+      });
+      return hasValidationErrors ? { ...acc, [id]: hasValidationErrors } : { ...acc };
+    }, {});
+
+    const hasEmptyRequiredInputs = !isNullOrEmpty(validationErrorsList);
```
Comment (rllyy97) [inline apps/designer-standalone/src/app/AzureLogicAppsDesigner/DesignerCommandBar.tsx:72]: We might want to rename this, it's not always empty for errors, could be a wrong type or something

Comment (ccastrotrejo) [reply apps/designer-standalone/src/app/AzureLogicAppsDesigner/DesignerCommandBar.tsx:72]: ohhh yeaah, thats true.  I will rename it. Thanks

**Comment by ccastrotrejo** [comment]
> Looks good, will you be adding this to Portal as well?

Yes, I will update portal with this.

**Pull request merged**